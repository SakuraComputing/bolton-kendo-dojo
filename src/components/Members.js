import React from 'react';
import { Link } from 'react-router-dom';

const  Members = () => {
  return (
    <div className="members">
      <h1 className="members-header">Welcome to the Members Area</h1>
      <div className="members-area-container">
        <div className="members-area-link">
          <Link to="/memberphoto">Photographs</Link>
          <p>Select the photgraphs to go in to the photgraph section</p>
        </div>
        <div className="members-area-link">
          <Link to="/memberphoto">Kendo Mind Map</Link>
          <p>Enter the Kendo Mind Map section for all deeper kendo study</p>
        </div>
        <div className="members-area-link">
          <Link to="/memberphoto">Videos</Link>
          <p>Go into the video section and view private members videos</p>
        </div>
        <div className="members-area-link">
          <Link to="/members">Bulletin Board</Link>
          <p>Add a post to the bulletin board. Responds to other posts</p>
        </div>
      </div>
    </div>
  )
}
export default Members;
