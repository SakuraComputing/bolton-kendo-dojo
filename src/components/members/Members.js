import React from 'react';
import { Link } from 'react-router-dom';

const  Members = () => {
  return (
    <div className="members">
      <h1 className="members-header">Welcome to the Members Area</h1>
      <div className="members-area-container">
        <div className="members-area-link">
          <Link to="/memberphoto">Photos</Link>
          <p>Select the photographs and upload these with text for other members to see.</p>
        </div>
        <div className="members-area-link">
          <Link to="/mindmap">Kendo Mind Map</Link>
          <p>Enter the Kendo Mind Map section for all deeper kendo study</p>
        </div>
        <div className="members-area-link">
          <Link to="/memberphoto">Videos</Link>
          <p>Go into the video section and view private members videos</p>
        </div>
        <div className="members-area-link">
          <Link to="/members">Bulletin Board</Link>
          <p>Add a post to the bulletin board. Respond to other posts</p>
        </div>
      </div>
    </div>
  )
}
export default Members;
