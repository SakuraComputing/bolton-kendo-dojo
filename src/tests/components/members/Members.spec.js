import React from 'react';
import { shallow } from 'enzyme';
import { Members } from '../../../components/members/Members.js';

describe('Members Page', () => {
    it('should render correctly', () => {
        const auth = { user: jest.fn }
        const members = shallow(<Members auth={auth} />)
        expect(members).toMatchSnapshot();
    })
})



