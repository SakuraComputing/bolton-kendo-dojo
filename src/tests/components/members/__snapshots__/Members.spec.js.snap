// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Members Page should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Members
    auth={
      Object {
        "user": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="members-header"
        >
          Welcome 
          bound fn
           to the Members Area
        </h1>,
        <div
          className="members-area-container"
        >
          <div
            className="members-area-link"
          >
            <Link
              replace={false}
              to="/memberphoto"
            >
              Photos
            </Link>
            <p>
              Select the photographs and upload these with text for other members to see.
            </p>
          </div>
          <div
            className="members-area-link"
          >
            <Link
              replace={false}
              to="/mindmap"
            >
              Kendo Mind Map
            </Link>
            <p>
              Enter the Kendo Mind Map section for all deeper kendo study
            </p>
          </div>
          <div
            className="members-area-link"
          >
            <Link
              replace={false}
              to="/memberphoto"
            >
              Videos
            </Link>
            <p>
              Go into the video section and view private members videos
            </p>
          </div>
          <div
            className="members-area-link"
          >
            <Link
              replace={false}
              to="/members"
            >
              Bulletin Board
            </Link>
            <p>
              Add a post to the bulletin board. Respond to other posts
            </p>
          </div>
        </div>,
      ],
      "className": "members",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Welcome ",
            "bound fn",
            " to the Members Area",
          ],
          "className": "members-header",
        },
        "ref": null,
        "rendered": Array [
          "Welcome ",
          "bound fn",
          " to the Members Area",
        ],
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="members-area-link"
            >
              <Link
                replace={false}
                to="/memberphoto"
              >
                Photos
              </Link>
              <p>
                Select the photographs and upload these with text for other members to see.
              </p>
            </div>,
            <div
              className="members-area-link"
            >
              <Link
                replace={false}
                to="/mindmap"
              >
                Kendo Mind Map
              </Link>
              <p>
                Enter the Kendo Mind Map section for all deeper kendo study
              </p>
            </div>,
            <div
              className="members-area-link"
            >
              <Link
                replace={false}
                to="/memberphoto"
              >
                Videos
              </Link>
              <p>
                Go into the video section and view private members videos
              </p>
            </div>,
            <div
              className="members-area-link"
            >
              <Link
                replace={false}
                to="/members"
              >
                Bulletin Board
              </Link>
              <p>
                Add a post to the bulletin board. Respond to other posts
              </p>
            </div>,
          ],
          "className": "members-area-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Link
                  replace={false}
                  to="/memberphoto"
                >
                  Photos
                </Link>,
                <p>
                  Select the photographs and upload these with text for other members to see.
                </p>,
              ],
              "className": "members-area-link",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Photos",
                  "replace": false,
                  "to": "/memberphoto",
                },
                "ref": null,
                "rendered": "Photos",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Select the photographs and upload these with text for other members to see.",
                },
                "ref": null,
                "rendered": "Select the photographs and upload these with text for other members to see.",
                "type": "p",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Link
                  replace={false}
                  to="/mindmap"
                >
                  Kendo Mind Map
                </Link>,
                <p>
                  Enter the Kendo Mind Map section for all deeper kendo study
                </p>,
              ],
              "className": "members-area-link",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Kendo Mind Map",
                  "replace": false,
                  "to": "/mindmap",
                },
                "ref": null,
                "rendered": "Kendo Mind Map",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Enter the Kendo Mind Map section for all deeper kendo study",
                },
                "ref": null,
                "rendered": "Enter the Kendo Mind Map section for all deeper kendo study",
                "type": "p",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Link
                  replace={false}
                  to="/memberphoto"
                >
                  Videos
                </Link>,
                <p>
                  Go into the video section and view private members videos
                </p>,
              ],
              "className": "members-area-link",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Videos",
                  "replace": false,
                  "to": "/memberphoto",
                },
                "ref": null,
                "rendered": "Videos",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Go into the video section and view private members videos",
                },
                "ref": null,
                "rendered": "Go into the video section and view private members videos",
                "type": "p",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Link
                  replace={false}
                  to="/members"
                >
                  Bulletin Board
                </Link>,
                <p>
                  Add a post to the bulletin board. Respond to other posts
                </p>,
              ],
              "className": "members-area-link",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Bulletin Board",
                  "replace": false,
                  "to": "/members",
                },
                "ref": null,
                "rendered": "Bulletin Board",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Add a post to the bulletin board. Respond to other posts",
                },
                "ref": null,
                "rendered": "Add a post to the bulletin board. Respond to other posts",
                "type": "p",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="members-header"
          >
            Welcome 
            bound fn
             to the Members Area
          </h1>,
          <div
            className="members-area-container"
          >
            <div
              className="members-area-link"
            >
              <Link
                replace={false}
                to="/memberphoto"
              >
                Photos
              </Link>
              <p>
                Select the photographs and upload these with text for other members to see.
              </p>
            </div>
            <div
              className="members-area-link"
            >
              <Link
                replace={false}
                to="/mindmap"
              >
                Kendo Mind Map
              </Link>
              <p>
                Enter the Kendo Mind Map section for all deeper kendo study
              </p>
            </div>
            <div
              className="members-area-link"
            >
              <Link
                replace={false}
                to="/memberphoto"
              >
                Videos
              </Link>
              <p>
                Go into the video section and view private members videos
              </p>
            </div>
            <div
              className="members-area-link"
            >
              <Link
                replace={false}
                to="/members"
              >
                Bulletin Board
              </Link>
              <p>
                Add a post to the bulletin board. Respond to other posts
              </p>
            </div>
          </div>,
        ],
        "className": "members",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Welcome ",
              "bound fn",
              " to the Members Area",
            ],
            "className": "members-header",
          },
          "ref": null,
          "rendered": Array [
            "Welcome ",
            "bound fn",
            " to the Members Area",
          ],
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="members-area-link"
              >
                <Link
                  replace={false}
                  to="/memberphoto"
                >
                  Photos
                </Link>
                <p>
                  Select the photographs and upload these with text for other members to see.
                </p>
              </div>,
              <div
                className="members-area-link"
              >
                <Link
                  replace={false}
                  to="/mindmap"
                >
                  Kendo Mind Map
                </Link>
                <p>
                  Enter the Kendo Mind Map section for all deeper kendo study
                </p>
              </div>,
              <div
                className="members-area-link"
              >
                <Link
                  replace={false}
                  to="/memberphoto"
                >
                  Videos
                </Link>
                <p>
                  Go into the video section and view private members videos
                </p>
              </div>,
              <div
                className="members-area-link"
              >
                <Link
                  replace={false}
                  to="/members"
                >
                  Bulletin Board
                </Link>
                <p>
                  Add a post to the bulletin board. Respond to other posts
                </p>
              </div>,
            ],
            "className": "members-area-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Link
                    replace={false}
                    to="/memberphoto"
                  >
                    Photos
                  </Link>,
                  <p>
                    Select the photographs and upload these with text for other members to see.
                  </p>,
                ],
                "className": "members-area-link",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Photos",
                    "replace": false,
                    "to": "/memberphoto",
                  },
                  "ref": null,
                  "rendered": "Photos",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Select the photographs and upload these with text for other members to see.",
                  },
                  "ref": null,
                  "rendered": "Select the photographs and upload these with text for other members to see.",
                  "type": "p",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Link
                    replace={false}
                    to="/mindmap"
                  >
                    Kendo Mind Map
                  </Link>,
                  <p>
                    Enter the Kendo Mind Map section for all deeper kendo study
                  </p>,
                ],
                "className": "members-area-link",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Kendo Mind Map",
                    "replace": false,
                    "to": "/mindmap",
                  },
                  "ref": null,
                  "rendered": "Kendo Mind Map",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Enter the Kendo Mind Map section for all deeper kendo study",
                  },
                  "ref": null,
                  "rendered": "Enter the Kendo Mind Map section for all deeper kendo study",
                  "type": "p",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Link
                    replace={false}
                    to="/memberphoto"
                  >
                    Videos
                  </Link>,
                  <p>
                    Go into the video section and view private members videos
                  </p>,
                ],
                "className": "members-area-link",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Videos",
                    "replace": false,
                    "to": "/memberphoto",
                  },
                  "ref": null,
                  "rendered": "Videos",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Go into the video section and view private members videos",
                  },
                  "ref": null,
                  "rendered": "Go into the video section and view private members videos",
                  "type": "p",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Link
                    replace={false}
                    to="/members"
                  >
                    Bulletin Board
                  </Link>,
                  <p>
                    Add a post to the bulletin board. Respond to other posts
                  </p>,
                ],
                "className": "members-area-link",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Bulletin Board",
                    "replace": false,
                    "to": "/members",
                  },
                  "ref": null,
                  "rendered": "Bulletin Board",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Add a post to the bulletin board. Respond to other posts",
                  },
                  "ref": null,
                  "rendered": "Add a post to the bulletin board. Respond to other posts",
                  "type": "p",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
